drake/systems/plants/collision/ResultCollector.h

```diff
@@ -0,0 +1,32 @@
+#ifndef DRAKE_SYSTEMS_PLANTS_COLLISION_RESULTCOLLECTOR_H_
+#define DRAKE_SYSTEMS_PLANTS_COLLISION_RESULTCOLLECTOR_H_
+
+#include "drake/systems/plants/collision/DrakeCollision.h"
+
+namespace DrakeCollision {
+Eigen::Vector3d toVector3d(const Eigen::Vector3d& vec);
+
+class ResultCollector {
+ public:
+  virtual ~ResultCollector() {}
+
+  virtual void addPointPairResult(const PointPair& result);
+
+  inline void addSingleResult(const ElementId idA, const ElementId idB,
+                              const Eigen::Vector3d& ptA,
+                              const Eigen::Vector3d& ptB,
+                              const Eigen::Vector3d& normal, double distance) {
```

Comment:
Value parameters shouldn't be marked `const` in a header in any case because that doesn't affect the API -- it is only relevant to the implementation. (As opposed to `const&` which *is* part of the API.)